docs/source/command_line.rst

```diff
@@ -1,4 +1,44 @@
 Executing via CLI
 =================
 
-TBD
+The package provides some basic utilities to simplify configuration and execution of pipelines via the command line. In
+the most basic case, one could use::
+
+
+  from dplutils.cli import cli_run
+
+  pipline = ...instantiate executor from graph...
+
+  if __name__ == "__main__":
+      cli_run(pipeline)
+
+
+This will setup an argument parser and add generic context and config setting arguments and call the `writeto` method
+writing to directory specified on the command line. It also has a help method::
+
+
+  python -m my_pipeline.main -h
+
+  usage: main.py [-h] [-c SET_CONTEXT] [-s SET_CONFIG] [-o OUT_DIR]
```

Comment:
Could you please clarify the relationship between `my_pipeline.main` and `main.py`. Is my_pipeline.main a python module?

Reply:
I'll clarify. Indeed that would be a python module that has an

```py
if __name__ == '__main__`
```

section (or is a module with `__main__.py` file. This would be where cli_run is executed.

Reply:
I clarified in text